main/src/com/google/refine/commands/row/AddRowsCommand.java

```diff
@@ -0,0 +1,77 @@
+
+package com.google.refine.commands.row;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.refine.commands.Command;
+import com.google.refine.model.AbstractOperation;
+import com.google.refine.model.Project;
+import com.google.refine.model.Row;
+import com.google.refine.operations.row.RowAdditionOperation;
+import com.google.refine.process.Process;
+import com.google.refine.util.Pool;
+
+public class AddRowsCommand extends Command {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        if (!hasValidCSRFToken(request)) {
+            respondCSRFError(response);
+            return;
+        }
+
+        try {
+            Project project = getProject(request);
+
+            String rowDataParam = "rows";
+            if (!request.getParameterMap().containsKey(rowDataParam) || request.getParameter(rowDataParam) == null) {
+                throw new ServletException(String.format("Parameter \"%s\" is required", rowDataParam));
+            }
+            String[] rowData = request.getParameterValues(rowDataParam);
+
+            // TODO: load Rows with existing pool. Current implementation requires users to re-reconcile values
+            //   in a previously reconciled column.
```

Comment:
I don't think it'd be too hard to address. But, I'm unsure how I'd deserialize a Row with an existing Pool. Recon is one of those areas of the code that I'm still don't know very well. As it stands now, a POST request to insert a row into a project with a reconciled column results in an unreconciled string in that column. The user then has to re-reconcile the column to address the new row, which I think is undesirable from a UI/UX perspective. There's also an edge case to consider where a user insert a new row where the recon entity is not in the existing pool. 

I'll work on it, but @wetneb can advise on how the best way to access an existing Pool in this scenario?

Reply:
Before we think about all that I would like to be clear on what is the use case (for an end user, using the web interface) behind supplying row data in this way. So far it's not clear to me how this is meant to be used. I think it would be simpler to concentrate on the feature of just adding blank rows first, for which this part is not needed.

If we do know we want to add rows with certain contents, determining how the user is meant to specify them (from the same project, duplicating them? from another project? input manually in some sort of dialog?) will help determine the right approach to specify this data in the API and therefore handle the recon part.

Reply:
If we're adding blank rows and the only unreconciled cells are the new blank ones, I think the current solution is fine. The wording of the TODO made me thing that the entire column would become unreconciled, which I don't believe is the case.

Reply:
I'd be in favour of not having support for loading non-blank rows in the back-end, if that feature isn't exposed in the UI and is known to be unfinished. That helps us avoid dead code.

Reply:
This demo that illustrates how new rows are typed as strings by default in a column that include reconciled cells. The same concept extends to other cell types: dates, numbers, null cells, booleans. 

https://github.com/OpenRefine/OpenRefine/assets/2933702/1c5bfc97-e4a3-4c23-9528-69defaebc9f2